src/server/scripts/Draenor/Auchindoun/boss_soulbinder_nyami.cpp

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (C) 2008-2016 TrinityCore <http://www.trinitycore.org/>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#include "ScriptMgr.h"
+#include "ScriptedCreature.h"
+
+enum Spells
+{
+    SPELL_MIND_SPIKE = 154415,
+    SPELL_SHADOW_WORD_PAIN = 154477,
+    SPELL_SOUL_VESSEL = 153477,
+    SPELL_TORN_SPIRITS = 153994,
+};
+
+enum Events
+{
+    EVENT_MIND_SPIKE,
```

Comment:
https://ideone.com/cUzTMl
You claim you tested all bosses in-game, yet you didn't notice that EVENT_MIND_SPIKE is never triggered.
Testing a script means adding a breakpoint to every event and ensuring it's executed.